app/kumactl/cmd/uninstall/uninstall_ebpf.go

```diff
@@ -0,0 +1,333 @@
+	cmd.Flags().DurationVar(&args.Timeout, "timeout", args.Timeout, "timeout for whole process of removing left files")
+	cmd.Flags().StringVar(&args.CleanupImageRegistry, "cleanup-image-registry", args.CleanupImageRegistry, "image registry for ebpf cleanup job")
+	cmd.Flags().StringVar(&args.CleanupImageRepository, "cleanup-image-repository", args.CleanupImageRepository, "image repository for ebpf cleanup job")
+	cmd.Flags().StringVar(&args.CleanupImageTag, "cleanup-image-tag", args.CleanupImageTag, "image tag for ebpf cleanup job")
+	cmd.Flags().StringVar(&args.CleanupJobName, "cleanup-job-name", args.CleanupJobName, "name of the cleanup job")
+	cmd.Flags().BoolVar(&args.RemoveOnly, "remove-only", args.RemoveOnly, "cleanup jobs and pods only")
+
+	return cmd
+}
+
+func genJobSpec(jobName, nodeName string, args *ebpfArgs) *batchv1.Job {
+	return &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: jobName,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						AppLabel: args.CleanupJobName,
+					},
+				},
+				Spec: corev1.PodSpec{
+					NodeName: nodeName,
+					Containers: []corev1.Container{
+						{
+							Name: args.CleanupJobName,
+							Image: fmt.Sprintf(
+								"%s/%s:%s",
+								args.CleanupImageRegistry,
+								args.CleanupImageRepository,
+								args.CleanupImageTag,
+							),
+							Command: []string{
+								"kumactl",
+								"uninstall",
+								"transparent-proxy",
+								"--ebpf-enabled",
+								"--ebpf-bpffs-path", args.BPFFsPath,
+							},
+							SecurityContext: &corev1.SecurityContext{
+								Privileged: new(bool),
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "bpf-fs-path",
+									MountPath: args.BPFFsPath,
+								},
+							},
+						},
+					},
+					RestartPolicy: corev1.RestartPolicyNever,
+					Volumes: []corev1.Volume{
+						{
+							Name: "bpf-fs-path",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: args.BPFFsPath,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *CleanupJob) getPodLogs(
+	ctx context.Context,
+	pod *corev1.Pod,
+) (string, error) {
+	podLogs, err := r.podClient.GetLogs(
+		pod.Name,
+		&corev1.PodLogOptions{},
+	).Stream(ctx)
+	if err != nil {
+		return "", fmt.Errorf("getting %s pod logs failed with error: %s",
+			pod.Name, err)
+	}
+	defer podLogs.Close()
+
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return "", fmt.Errorf("copy data from podLogs to buf failed with error:"+
+			" %s", err)
+	}
+
+	return buf.String(), nil
+}
+
+func (r *CleanupJob) Watch(ctx context.Context, watcher watch.Interface) error {
+	eg, _ := errgroup.WithContext(ctx)
+
+	eg.Go(func() error {
+		r.Lock()
+		defer r.Unlock()
```

Comment:
what is accessed concurrently here?

Reply:
good catch, I think we are updating map but it doesn't seem to be concurrently